Plugins/Object/Object.cpp

```diff
@@ -594,4 +598,72 @@ ArgumentStack Object::SetTriggerGeometry(ArgumentStack&& args)
     return stack;
 }
 
+ArgumentStack Object::RemoveIconEffect(ArgumentStack&& args)
+{
+    ArgumentStack stack;
+
+    if (auto *pObject = object(args))
+    {
+        const auto nIcon = Services::Events::ExtractArgument<int32_t>(args);
+
+        for (int i = 0; i < pObject->m_appliedEffects.num; i++)
+        {
+            auto *eff = pObject->m_appliedEffects.element[i];
+
+            if (eff->m_sCustomTag == "NWNX_Object_IconEffect" && eff->m_nParamInteger[0] == nIcon)
+            {
+                pObject->RemoveEffect(eff);
```

Comment:
Since these icon effects are unique, you can add a `break;` here to exit the for loop when the effect has been found